autoload/airline/extensions/branch.vim

```diff
@@ -3,11 +3,14 @@
 
 scriptencoding utf-8
 
+" TODO: Checking for autoload functions is a bad idea. exists checks don't
+" work for autoload functions unless they happen to be loaded before us.
 let s:has_fugitive = exists('*fugitive#head')
 let s:has_lawrencium = exists('*lawrencium#statusline')
 let s:has_vcscommand = get(g:, 'airline#extensions#branch#use_vcscommand', 0) && exists('*VCSCommandGetStatusLine')
+let s:has_custom_scm = exists('airline#extensions#branch#custom_head')
 
-if !s:has_fugitive && !s:has_lawrencium && !s:has_vcscommand
+if !s:has_fugitive && !s:has_lawrencium && !s:has_vcscommand && !s:has_custom_scm
```

Comment:
I would rather suggest to use: `!empty(get(g:, 'airline#extensions#branch#custom_head', ''))`

Reply:
But the file isn't sourced if any of these conditions are false (line 14). So even if I followed that suggestion, I don't see how it would make a difference. Adding custom_head after first source would not make it start working. Re-sourcing the file would make it start working with or without changing this line.

The same logic follows for the other two cases of this.

I think it makes more sense to take my change and then refactor the file to do these checks for all scm types.